### PR TITLE
Extension of T2T Ksonnet component

### LINKED ISSUE #147: [code_search] t2t-datagen on a Kubeflow cluster
**NOTE**: Splitting isolated tasks from #121 .

/area example/code_search

### LINKED ISSUE #148: [code_search] t2t-exporter on a Kubeflow cluster
This model export will allow for usage by TF Serving.

**NOTE**: Splitting isolated tasks from #121.

/area example/code_search

### PR DESCRIPTION
**Pull request by activatedgeek**
This PR tests out and extends the base Ksonnet component for Tensor2Tensor based jobs.

Currently, everything is being tested out with GCP based credentials.

Fixes #147, Fixes #148 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/149)

### CODE REVIEW

**Comment by activatedgeek** [comment]
This blocked on https://github.com/tensorflow/tensor2tensor/issues/879

**Comment by activatedgeek** [comment]
A workaround implemented right now is to download the data locally to use `gsutil` (if needed, and upload back on completion) and send in `bash` command to the container using (`-c`). It is a simple wrap function around the existing code so removing it should be trivial in future.

File: code_search/create_secrets.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
```
Comment (ankushagarwal) [inline code_search/create_secrets.sh]: Can you add a short docstring here to explain what this does?

**Pull request merged**